src/mad/sh/scripts_common.sh

```diff
@@ -577,25 +609,63 @@ set -e -o pipefail
 if [ ! -d $2 ]; then
     mkdir -p $2
 fi
+EOF`
+    SSH_EXEC_RC=$?
 
-# create or update .monitor content
-if [ -n "$3" ]; then
-    MONITOR_FN="\$(dirname $2)/.monitor"
+    if [ $SSH_EXEC_RC -ne 0 ]; then
+        error_message "Error creating directory $2 at $1: $SSH_EXEC_ERR"
 
-    MONITOR=''
-    if [ -f "\\${MONITOR_FN}" ]; then
-        MONITOR="\\$(cat "\\${MONITOR_FN}" 2>/dev/null)"
+        exit $SSH_EXEC_RC
     fi
+}
 
-    if [ "x\\${MONITOR}" != "x$3" ]; then
-        echo "$3" > "\\${MONITOR_FN}"
-    fi
+# Enables local monitoring for host $2 and DS $1, by creating the corresponding
+# .monitor file containing the $3 ( the driver name) (e.g ssh)
+function enable_local_monitoring
+{
+    SSH_EXEC_ERR=`$SSH $1 bash -s 2>&1 1>/dev/null <<EOF
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Use $(...) notation instead of legacy backticks `...`.